pymc/distributions/multivariate.py

```diff
@@ -1143,73 +1117,115 @@ def _random(self, n, eta, size=1):
+        Calculate log-probability of Covariance matrix with LKJ
+        distributed correlations at specified value.
 
         Parameters
         ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
+        value: numeric
+            Value for which log-probability is calculated.
 
         Returns
         -------
-        array
+        TensorVariable
         """
-        # # Get parameters and broadcast them
-        # n, eta = draw_values([self.n, self.eta], point=point, size=size)
-        # broadcast_shape = np.broadcast(n, eta).shape
-        # # We can only handle cov matrices with a constant n per random call
-        # n = np.unique(n)
-        # if len(n) > 1:
-        #     raise RuntimeError("Varying n is not supported for LKJCholeskyCov")
-        # n = int(n[0])
-        # dist_shape = ((n * (n + 1)) // 2,)
-        # # We make sure that eta and the drawn n get their shapes broadcasted
-        # eta = np.broadcast_to(eta, broadcast_shape)
-        # # We change the size of the draw depending on the broadcast shape
-        # sample_shape = broadcast_shape + dist_shape
-        # if size is not None:
-        #     if not isinstance(size, tuple):
-        #         try:
-        #             size = tuple(size)
-        #         except TypeError:
-        #             size = (size,)
-        #     if size == sample_shape:
-        #         size = None
-        #     elif size == broadcast_shape:
-        #         size = None
-        #     elif size[-len(sample_shape) :] == sample_shape:
-        #         size = size[: len(size) - len(sample_shape)]
-        #     elif size[-len(broadcast_shape) :] == broadcast_shape:
-        #         size = size[: len(size) - len(broadcast_shape)]
-        # # We will always provide _random with an integer size and then reshape
-        # # the output to get the correct size
-        # if size is not None:
-        #     _size = np.prod(size)
-        # else:
-        #     _size = 1
-        # samples = self._random(n, eta, size=_size)
-        # if size is None:
-        #     samples = samples[0]
-        # else:
-        #     samples = np.reshape(samples, size + sample_shape)
-        # return samples
 
-    def _distr_parameters_for_repr(self):
-        return ["eta", "n"]
+        diag_idxs = at.cumsum(at.arange(1, n + 1)) - 1
+        cumsum = at.cumsum(value ** 2)
+        variance = at.zeros([n])
+        variance = at.inc_subtensor(variance[0], value[0] ** 2)
+        variance = at.inc_subtensor(variance[1:], cumsum[diag_idxs[1:]] - cumsum[diag_idxs[:-1]])
+        sd_vals = at.sqrt(variance)
+
+        # TODO: Since the sd_dist logp is added independently,
+        # we can perhaps compose the logp terms in a more clean way
```

Comment:
The issue with calling `logpt` directly over here is that `logpt` tries to filter the input values according to RV's shape. So we'd have to put an limitation on shapes if we try to do it like that.

Reply:
I didn't mean that. But this approach seems fine in any case. As long as it doesn't result in some value variable missing warnings